pyembed/src/extension.rs

```diff
@@ -143,114 +135,43 @@ fn register_pkg_resources(py: cpython::Python) -> cpython::PyResult<cpython::PyO
 /// This receives a handle to the current Python interpreter and just-created
 /// Python module instance. It populates the internal module state and registers
 /// functions on the module object for usage by Python.
-fn module_init(py: cpython::Python, m: &cpython::PyModule) -> cpython::PyResult<()> {
+fn module_init(py: Python, m: &PyModule) -> PyResult<()> {
     // Enforce minimum Python version requirement.
     //
     // Some features likely work on older Python versions. But we can't
     // guarantee it. Let's prevent footguns.
     let sys_module = py.import("sys")?;
-    let version_info = sys_module.get(py, "version_info")?;
+    let version_info = sys_module.getattr("version_info")?;
 
-    let major_version = version_info.getattr(py, "major")?.extract::<i32>(py)?;
-    let minor_version = version_info.getattr(py, "minor")?.extract::<i32>(py)?;
+    let major_version = version_info.getattr("major")?.extract::<i32>()?;
+    let minor_version = version_info.getattr("minor")?.extract::<i32>()?;
 
     if major_version < 3 || minor_version < 8 {
```

Comment:
You can use https://docs.rs/pyo3/0.14.2/pyo3/prelude/struct.Python.html#method.version_info here:

```suggestion
    py.version_info() <= (3, 8) {
```